qemu/tests/qmp_basic.py

```diff
@@ -332,7 +335,10 @@ def argument_checker_suite(monitor):
 
         # value argument must be a json-number
         for arg in ({}, [], True, "foo"):
-            resp = monitor.cmd_qmp("migrate_set_speed", {"value": arg})
+            if utils_misc.compare_qemu_version(5, 1, 0) is True:
+                resp = monitor.cmd_qmp("migrate-set-parameters", {"value": arg})
+            else:
+                resp = monitor.cmd_qmp("migrate_set_speed", {"value": arg})
             check_error_resp(resp, "GenericError",
```

Comment:
Go through the function `check_error_resp`, it is a bug that `datadict` never be used, means the setting `{"name": "value", "expected": "number"}` is meaningful.

This is a serious bug cause missing one importance test scenario.

Reply:
Please consider this problem, otherwise the whole change is useless IMO

Reply:
> Go through the function `check_error_resp`, it is a bug that `datadict` never be used, means the setting `{"name": "value", "expected": "number"}` is meaningless.
> 
> This is a serious bug cause missing one important test scenario.

@yanan-fu 
Yes, The missing is the "desc" part,  I already have updated it, please help to review latest patch? thanks